src/Cpanel/ClientActions.php

```diff
@@ -32,6 +33,19 @@ public function __construct(DefaultIntegration $cpanelIntegration, APIInterface
         $this->logger = $cpanelIntegration->getLogger();
         $this->partialZoneSet = new Partial($this->cpanelAPI, $this->dataStore, $this->logger);
         $this->request = $request;
+
+        $this->setPunycoder();
+    }
+
+    public function setPunycoder($p = null) {
+        $this->punyCode = $p;
```

Comment:
Normally we put this logic in the getter but it doesn't matter, this approach works as well.